app/models/proof_attempt.rb

```diff
@@ -1,6 +1,13 @@
 class ProofAttempt < ActiveRecord::Base
   belongs_to :theorem, foreign_key: 'sentence_id'
   belongs_to :proof_status
+  has_and_belongs_to_many :used_axioms,
```

Comment:
Rails/HasAndBelongsToMany: Prefer `has_many :through` to `has_and_belongs_to_many`.